src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/CodeGen/ReadyToRunObjectWriter.cs

```diff
@@ -120,16 +132,31 @@ public void EmitPortableExecutable()
                 }
 
                 r2rPeBuilder.SetCorHeader(_nodeFactory.CopiedCorHeaderNode, _nodeFactory.CopiedCorHeaderNode.Size);
+                r2rPeBuilder.SetDebugDirectory(_nodeFactory.DebugDirectoryNode, _nodeFactory.DebugDirectoryNode.Size);
 
                 if (_nodeFactory.Win32ResourcesNode != null)
                 {
                     Debug.Assert(_nodeFactory.Win32ResourcesNode.Size != 0);
                     r2rPeBuilder.SetWin32Resources(_nodeFactory.Win32ResourcesNode, _nodeFactory.Win32ResourcesNode.Size);
                 }
 
+                var memStream = new MemoryStream();
```

Comment:
Ok let me look into it. I wanted to avoid having to do this with files to not get into any read/write locking issues that could cause potential failures

Reply:
I don't mean closing the file stream - the file stream should stay open - we just seek in it and read from it. That shouldn't hit any locking issues (it's also what crossgen does there if I'm reading it right)

Reply:
Yep. I wasn't sure it was possible to use the stream for reading too, but turns out that works pretty well. Fixed